pkg/registry/apis/wireset.go

```diff
@@ -15,10 +16,11 @@ import (
 var WireSet = wire.NewSet(
 	ProvideRegistryServiceSink, // dummy background service that forces registration
 
-	wire.Bind(new(datasource.QuerierProvider), new(*datasource.DefaultQuerierProvider)),
-	datasource.ProvideDefaultQuerierProvider,
+	// read-only datasource abstractions
 	plugincontext.ProvideService,
-	wire.Bind(new(datasource.PluginContextProvider), new(*plugincontext.Provider)),
+	wire.Bind(new(datasource.PluginContextWrapper), new(*plugincontext.Provider)),
+	//datasource.ProvideDefaultPluginConfigs,
+	alt.ProvideSQLBasedPluginConfigs, // only works for testdata now
```

Comment:
^^^ do not merge, but worth showing the alternative approach works.  With this approach a standalone version can be pointed to a different storage approach easily